R/weight.R

```diff
@@ -41,33 +43,216 @@ weight <- function (x) {
     currentWeight <- ShojiEntity(crGET(shojiURL(x, "fragments", "preferences")))$weight
     if (!identical(value, currentWeight)) {
         crPATCH(shojiURL(x, "fragments", "preferences"),
-            body=toJSON(list(weight=value)))
+            body = toJSON(list(weight = value)))
         x <- refresh(x)
     }
     return(x)
 }
 
 #' @rdname weight
 #' @export
-setMethod("weightVariables", "CrunchDataset",
-    function (x) weightVariables(allVariables(x)))
+is.weight <- function(x){
+    if (is.variable(x)) {
+        ds <- loadDataset(datasetReference(self(x)))
+        return(identical(weight(ds), x))
+    } else {
+        return(FALSE)
+    }
+}
 
 #' @rdname weight
 #' @export
-setMethod("weightVariables", "VariableCatalog", function (x) {
+setMethod("is.weight<-", "NumericVariable", function(x, value) {
+    if (!(is.logical(value)  && length(value) == 1)) {
```

Comment:
Don't put two spaces before `&&` (I saw another case of that somewhere else). Also, this check allows `NA` (which is `logical` length 1) to go through.

Reply:
I think the conjunction is negated right? So it doesn't go through: 
```
 > value <- NA
 > !(is.logical(value) && length(value) == 1)
[1] FALSE
```
Is there a clearer way of doing a NAND gate?